src/marqo/s2_inference/processing/image.py

```diff
@@ -206,20 +208,21 @@ def _load_and_cache_model(self):
         model_type = (self.model_name, self.device)
         model_cache_key = _create_model_cache_key(self.model_name, self.device)
 
-        if model_cache_key not in get_available_models():
-            logger.info(f"loading model {model_type}")
-            if model_type[0] in self.allowed_model_types:
-                func = self.model_load_function
-            else:
-                raise TypeError(f"wrong model for {model_type}")
+        with load_model_lock:
```

Comment:
Should you raise an error here if the lock is not acquired? Or are you going to keep waiting?

Reply:
I think we can put a timeout error if we wait too long. I'm not sure if an error if necessary if a lock is not acquired immediately

Reply:
In `s2_inference.vectorise`, a conflict error is raised directly. I think we should follow the same manner.